### PR TITLE
Build tests as SDK projects

### PR DESCRIPTION
**Pull request by echesakovMSFT**
This PR will be easier to review in commit-by-commit basis. Part of the PR is done manually while 99% of the changes are automatic.

cf56dc1939: Remove ReferenceSystemPrivateCoreLib in tests/src/JIT/Directed/arglist/vararg.csproj

05c0db7eb0: Don't generate Bash execution script when building on Windows in tests/src/CLRTest.Execute.targets

44b00e5d4d: Add Microsoft.NET.Sdk.IL in global.json

73dda7abd8: Set Platform to BuildArch in tests/dir.sdkbuild.props

5ea9396e81: Don't override $(TargetExt) in tests/src/dir.targets Pt.1

4c9a7e8274: Don't override $(TargetExt) in tests/src/dir.targets Pt.2

93802c8fe2: Remove PointerSize in tests/src/dir.props and move the logic in the only user in tests/src/JIT/Regression/JitBlue/DevDiv_278523/DevDiv_278523.ilproj

c096cd4fd2: Fix issue with resource loading in tests/src/JIT/Performance/CodeQuality/BenchmarksGame tests/src/Regressions/coreclr/GitHub_22888/test22888.cs

4115e2ca3f: Set $(DisableImplicitFrameworkReferences) when $(ReferenceSystemPrivateCoreLib) is true in tests/src/dir.targets

55e69b0baa: [TEST DISABLED] tests/src/reflection/GenericAttribute/GenericAttributeTests.csproj

02e6b9de5d: [TEST DISABLED] Interop/WinRT/NETClients/Bindings/NETClientBindings/NETClientBindings.csproj

6d6c7681d3: Rename dir.props -> Directory.Build.props under tests\src\JIT

288bd6e328: Fix typo in tests/dir.common.props MSBuildProjectDirectary -> MSBuildProjectDirectory

42fabcd9f4: Remove ResetReferenceCopyLocalPaths Target in tests/src/dir.targets

cd5657a30f: Disable GenerateDependencyFile and GenerateRuntimeConfigurationFiles in tests/src/dir.props

0cec3e2479: Remove AutoGenerateBindingRedirects logic

57336d55cf: Update tests\src\sizeondisk\sodbench\SoDBench.csproj

4f72449985: Rename dir.targets -> Directory.Build.targets under tests\src\GC\Stress\Tests

99a10e0573: Rename *.sc -> *.cs under tests\src\JIT\Regression\CLR-x86-JIT\V1-M10

151e1c163f: Specify Link="%(Filename)%(Extension)" so vcruntime*d.dll and msvcp*.dll copied directly to the output directory

**Automatic changes start HERE**

5f6ae843d4: * Remove unnecessary comments

* Remove default configurations

* Remove MsBuildProjectDirOverride

* Remove CodeAnalysisDependentAssemblyPaths

* Remove Service added by VS

6ed1e28f72: Remove unnecessary properties

2b88aeef76: * Remove Project attributes

* Set Project.Sdk attribute value

cd05a417dd: Remove Import elements

61bdf20f78: Clean-up ProjectReferences to CoreCLRTestLibrary.csproj

da2a84cd39: Remove default values of $(AssemblyName), $(RootNamespace), $(FileAlignment)

b5d477e847: Transform items to Compile with either Include=$(MSBuildProjectName).cs or Include=$(MSBuildProjectName).il when

1) It is the only Compile Item in this project;

2) $(MSBuildThisFileDirectory), $(MSBuildProjectName) and the Compile item filename are the same.

4708b582fe: Remove empty PropertyGroups and ItemGroups

4e6fb35b2d: Remove all Xml namespaces

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
:shipit:

File: global.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "Microsoft.DotNet.Arcade.Sdk": "1.0.0-beta.19372.10",
     "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19372.10",
     "Microsoft.Build.NoTargets": "1.0.53",
-    "Microsoft.Build.Traversal": "2.0.2"
+    "Microsoft.Build.Traversal": "2.0.2",
+    "Microsoft.NET.Sdk.IL": "3.0.0-preview1-27101-02"
```
Comment (BruceForstall) [inline global.json]: Where does this version come from? Is it updated automatically somehow? Doesn't look like it's part of Versions.props for example.

File: tests/src/CoreMangLib/system/delegate/delegate/DelegateGetInvocationList1.csproj [hunk 2]
```diff
@@ -1,40 +1,14 @@
...
-  </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="$(SourceDir)Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj" />
   </ItemGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-  <PropertyGroup Condition=" '$(MsBuildProjectDirOverride)' != '' ">
-  </PropertyGroup>
-</Project>
-
+</Project>
```
Comment (BruceForstall) [inline tests/src/CoreMangLib/system/delegate/delegate/DelegateGetInvocationList1.csproj]: It looks like the modified project files don't have a trailing CR/LF; just a trailing null on the last line. Seems like they should have the CR/LF.

File: global.json (hunk 1 above)
Comment (echesakovMSFT) [reply global.json]: I looked in dotnet/corefx and it seems that `Microsoft.NET.Sdk.IL` can be update automatically by dotnet-maestro bot (e.g. https://github.com/dotnet/corefx/commit/739f4437298d3d010f3427f557bf6d189aac67a7). I will figure out how to do the same in dotnet/coreclr after I merge this change.

File: tests/src/CoreMangLib/system/delegate/delegate/DelegateGetInvocationList1.csproj (hunk 2 above)
Comment (echesakovMSFT) [reply tests/src/CoreMangLib/system/delegate/delegate/DelegateGetInvocationList1.csproj]: Done.

File: tests/dir.sdkbuild.props [hunk 3]
```diff
@@ -14,6 +14,9 @@
    
     <!-- Force the CLI to allow us to target higher netcoreapp than it may know about -->
     <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
+
+    <!-- TODO: remove this when we use the Arcade Sdk. -->
+    <DeterministicSourcePath>false</DeterministicSourcePath>
```
Comment (AaronRobinsonMSFT) [inline tests/dir.sdkbuild.props]: Please update the comment to contain `[ARCADE]`.

**Comment by echesakovMSFT** [comment]
I had to *explicitly* disable `DeterministicSourcePaths` since we now specify `ContiniousIntegrationBuild` which forces Roslyn to run in Deterministic mode. To be able to use this mode we need to be on Arcade Sdk (and we are not at the moment) so SourceLink can determine the SourceRoot. If left enabled it will cause the following error during the test build
```
##[error].packages\microsoft.net.compilers.toolset\3.2.0-beta2-19272-03\tasks\netcoreapp2.1\Microsoft.Managed.Core.targets(102,5): error : SourceRoot items must include at least one top-level (not nested) item when DeterministicSourcePaths is true
```
As soon as we get out of BuildTools and switch to Arcade Sdk this won't be needed anymore. 

If no one has any concerns with this particular change I will merge the PR as soon as outerloop testing passes.

**Comment by echesakovMSFT** [comment]
Don't see any suspicious failures in coreclr-jitstress. coreclr-ci and coreclr-outerloop are both clean.

**Pull request merged**